src/PKSim.Core/Services/SimulationToProjectSnapshotMapper.cs

```diff
@@ -0,0 +1,38 @@
+using System;
+using System.Linq;
+using System.Text;
+using OSPSuite.Core.Domain;
+using OSPSuite.Utility;
+using OSPSuite.Utility.Extensions;
+using PKSim.Core.Model;
+using PKSim.Core.Snapshots.Mappers;
+using ModelSimulation = PKSim.Core.Model.Simulation;
+
+namespace PKSim.Core.Services;
+
+public interface ISimulationToProjectSnapshotMapper : IMapper<ModelSimulation, string>;
+
+public class SimulationToProjectSnapshotSnapshotMapper(
+   IJsonSerializer jsonSerializer,
+   ISnapshotMapper snapshotMapper,
+   IPKSimProjectRetriever pkSimProjectRetriever) : ISimulationToProjectSnapshotMapper
+{
+   public string MapFrom(ModelSimulation simulation)
+   {
+      var pkSimProject = createProjectFrom(simulation);
+      var projectSnapshot = snapshotMapper.MapToSnapshot(pkSimProject).Result;
+      return Convert.ToBase64String(Encoding.UTF8.GetBytes(jsonSerializer.Serialize(projectSnapshot)));
+   }
+
+   private PKSimProject createProjectFrom(ModelSimulation simulation)
```

Comment:
The easiest way to export everything needed for a simulation is to create a project that contains everything needed for the simulation and nothing more, then create a snapshot of the project.